platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractCollectionProperty.java

```diff
@@ -600,4 +606,82 @@ public void replace(Transformer<? extends @org.jetbrains.annotations.Nullable Pr
             set((Iterable<? extends T>) null);
         }
     }
+
+    @Override
+    public CompoundAssignmentStandIn toCompoundOperand() {
+        return new CompoundAssignmentStandIn();
+    }
+
+    public class CompoundAssignmentStandIn {
```

Comment:
💅 I'd add a note that this class contains methods called by Groovy under the hood